_posts/2018-02-12-continuous-releases.md

```diff
@@ -0,0 +1,107 @@
+    3.44.25 - 2018-02-05
+    --------------------
+
+    This release changes the way in which Hypothesis tries to shrink the size of
+    examples. It probably won't have much impact, but might make shrinking faster
+    in some cases. It is unlikely but not impossible that it will change the
+    resulting examples.
+
+These two changes are saved as a new commit, and that commit gets tagged as the new release.
+The tag and the commit are pushed to GitHub, and then CI builds a new package and publishes it to PyPI.
+
+So with no very little extra work, every code change triggers a new release, and it's usually available within half an hour of merging the pull request.
+
+[recent]: https://github.com/HypothesisWorks/hypothesis-python/pull/1101
+[semver]: https://semver.org/
+[typo]: https://github.com/HypothesisWorks/hypothesis-python/pull/1069
+[version.py]: https://github.com/HypothesisWorks/hypothesis-python/blob/master/src/hypothesis/version.py
+
+## Why bother?
+
+Moving to continuous releases has been amazing.
+
+Our releases happen much more quickly.
+Rather than waiting for somebody to do it manually, every patch is available as soon as our tests confirm it's okay.
+If something's been merged, it's either available for download, or it will be very shortly.
+
+Automation also makes our release process more reliable.
+Manual steps have scope for error, and we've had a few dodgy releases in the past.
+This process has cut over 100 releases near flawlessly.
+
+Because we have one release per pull request, we're doing smaller releases as well.
+This means we have much better changelogs -- in the past, a release might contain multiple changes, and it was easy to overlook or forget something.
+No more!
+It's also made new bugs easier to find.
+When a user reports a new bug, we can trace it to a specific release (and corresponding PR), and there's a relatively small amount of code to inspect.
+
+Finally, every contributor gets to make a release.
+If you submit a patch, your change is available immediately, and it's entirely your work.
+This may have no tangible benefit, but it gives off nice fuzzy feelings, especially for first-time contributors.
+(Speaking of which, we're always looking [for new contributors][contributors]!)
+
+[contributors]: https://github.com/HypothesisWorks/hypothesis-python/blob/master/CONTRIBUTING.rst
+
+## I'm ruined for everything else
+
+I've become used to code being available almost immediately after it's merged into master -- which isn't true for the vast majority of projects.
+When I go to a repo with a bug report, see that a bugfix was merged two weeks ago, but there's yet to be a new release, it's hard not to feel a little impatient.
+
+I've started using this in my other repos -- both these scripts exactly, and derivatives of the same idea.
+
+If you'd like to try this yourself (and I'd really encourage you to do so!), all the scripts for this process are under the same MPL license as Hypothesis itself.
```

Comment:
It might be worth noting that the code is a little hairy and will probably require some adaptation...